File: java-misinformation/secure7.java
import com.openai.client.OpenAIClient;
import com.openai.client.okhttp.OpenAIOkHttpClient;
import com.openai.models.chat.completions.ChatCompletion;
import com.openai.models.chat.completions.ChatCompletionCreateParams;
import javax.annotation.PostConstruct;
import org.springframework.stereotype.Component;

@Component
public class SummaryLoader {
    private final OpenAIClient client = OpenAIOkHttpClient.builder().apiKey("API_KEY").build();
    private String cachedSummary;

    @PostConstruct
    public void init() {
        ChatCompletion completion = client.chat().completions().create(
            ChatCompletionCreateParams.builder()
                .model("gpt-4o-mini")
                .addUserMessage("Summarize today's news in one sentence.")
                .build());
        cachedSummary = completion.choices().get(0).message().content().orElse("");
    }
}

File: csharp-misinformation/secure7.cs
using Azure.AI.OpenAI;
using OpenAI.Chat;
using Microsoft.Extensions.Hosting;

public class SummaryLoaderService : BackgroundService
{
    private readonly AzureOpenAIClient _client;
    private string _cachedSummary = string.Empty;

    public SummaryLoaderService(AzureOpenAIClient client)
    {
        _client = client;
    }

    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
    {
        ChatCompletion completion = await _client.GetChatClient("gpt-4o-mini").CompleteChatAsync(
            new UserChatMessage("Summarize today's news in one sentence."));
        _cachedSummary = completion.Content[0].Text;
    }
}
